### PR TITLE
chore: Improved native options checks and logging

### PR DESCRIPTION
**Pull request by bitsandfoxes**
It was getting a bit hard to follow what was enabled/disabled and why.

#skip-changelog

### CODE REVIEW

File: src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs [hunk 1]
```diff
@@ -147,38 +149,51 @@ internal void ModifyManifest(string basePath)
...
         {
-            var options = _getOptions();
-            var logger = options?.DiagnosticLogger ?? new UnityLogger(new SentryUnityOptions());
+            var disableSymbolsUpload = false;
+            var logger = _options?.DiagnosticLogger ?? new UnityLogger(new SentryUnityOptions());
+            var symbolsUpload = new DebugSymbolUpload(logger, unityProjectPath, gradleProjectPath,
+                EditorUserBuildSettings.exportAsGoogleAndroidProject);
 
-            if (_scriptingImplementation != ScriptingImplementation.IL2CPP)
+            if (_options is null)
```
Comment (vaind) [inline src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs:157]:
```suggestion
            if (_options is null || !_options.Enabled || !_options.AndroidNativeSupportEnabled)
```

File: src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs [hunk 2]
```diff
@@ -147,38 +149,51 @@ internal void ModifyManifest(string basePath)
...
-                logger.LogDebug("Automated symbols upload requires the IL2CPP scripting backend.");
-                return;
+                disableSymbolsUpload = true;
+            }
+            else if (!_options.Enabled)
+            {
+                disableSymbolsUpload = true;
+            }
+            else if (!_options.AndroidNativeSupportEnabled)
+            {
```
Comment (vaind) [inline src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs:166]:
```suggestion
```

File: src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs [hunk 3]
```diff
@@ -147,38 +149,51 @@ internal void ModifyManifest(string basePath)
...
             {
                 logger.LogWarning("Failed to load sentry-cli options.");
-                symbolsUpload.RemoveUploadFromGradleFile();
-
-                return;
+                disableSymbolsUpload = true;
+            }
+            else if (!_sentryCliOptions.IsValid(logger, _isDevelopmentBuild))
+            {
+                disableSymbolsUpload = true;
```
Comment (vaind) [inline src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs:177]: maybe a log so that users get at least some feedback that symbols are not uploaded and (kinda) why?

**Review by vaind — APPROVED** [review]

File: src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs (hunk 3 above)
Comment (bitsandfoxes) [reply src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs:169]: The "why" logs come from within `IsValid`.
